From f6b01e19decf91b4f0785be1337520c2ebcefc20 Mon Sep 17 00:00:00 2001
From: Connor McAdams <conmanx360@gmail.com>
Date: Fri, 4 Mar 2022 22:06:27 -0500
Subject: [PATCH 02/10] ALSA: hda/ca0132 - Add SoundBlaster AE-9 pincfg.

Add AE-9 pincfg, based on the values set within Windows.

Signed-off-by: Connor McAdams <conmanx360@gmail.com>
---
 sound/pci/hda/patch_ca0132.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/sound/pci/hda/patch_ca0132.c b/sound/pci/hda/patch_ca0132.c
index 5962545a0d4a..5b08807fab54 100644
--- a/sound/pci/hda/patch_ca0132.c
+++ b/sound/pci/hda/patch_ca0132.c
@@ -1294,6 +1294,19 @@ static const struct hda_pintbl ae7_pincfgs[] = {
 	{}
 };
 
+static const struct hda_pintbl ae9_pincfgs[] = {
+        { 0x0b, 0x01017010 },
+        { 0x0c, 0x414510f0 },
+        { 0x0e, 0x01c520f0 },
+        { 0x0f, 0x01017114 },
+        { 0x10, 0x01017011 },
+        { 0x11, 0x41a170ff },
+        { 0x12, 0x41a170f0 },
+        { 0x13, 0x908700f0 },
+        { 0x18, 0x500000f0 },
+	{}
+};
+
 static const struct snd_pci_quirk ca0132_quirks[] = {
 	SND_PCI_QUIRK(0x1028, 0x057b, "Alienware M17x R4", QUIRK_ALIENWARE_M17XR4),
 	SND_PCI_QUIRK(0x1028, 0x0685, "Alienware 15 2015", QUIRK_ALIENWARE),
@@ -9757,6 +9770,10 @@ static void ca0132_config(struct hda_codec *codec)
 		codec_dbg(codec, "%s: QUIRK_AE7 applied.\n", __func__);
 		snd_hda_apply_pincfgs(codec, ae7_pincfgs);
 		break;
+	case QUIRK_AE9:
+		codec_dbg(codec, "%s: QUIRK_AE9 applied.\n", __func__);
+		snd_hda_apply_pincfgs(codec, ae9_pincfgs);
+		break;
 	default:
 		break;
 	}
@@ -9839,6 +9856,7 @@ static void ca0132_config(struct hda_codec *codec)
 		break;
 	case QUIRK_AE5:
 	case QUIRK_AE7:
+	case QUIRK_AE9:
 		spec->num_outputs = 2;
 		spec->out_pins[0] = 0x0B; /* Line out */
 		spec->out_pins[1] = 0x11; /* Rear headphone out */
-- 
2.43.0

